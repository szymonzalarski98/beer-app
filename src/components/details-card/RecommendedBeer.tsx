import React from 'react';
import { TouchableOpacity, View, Text, Image, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { color, fontSize, deviceWidth } from '../../styles/helpers.style';
import Property from '../Property';

interface Props {
  id: number;
  name: String;
  abv: number;
  ibu: number;
  image_url: string;
}

const RecommendedBeer: React.SFC<Props> = (props) => {
  const { id, name, abv, ibu, image_url } = props;
  const navigation = useNavigation();
  return (
    <TouchableOpacity onPress={() => navigation.push('Details', { id })} style={styles.container}>
      {
        image_url !== null
        ?
        <Image source={{ uri: image_url }} style={styles.image} resizeMode="contain" />
        :
        <View style={styles.blankImage} />
      }
      <View style={styles.infoContainer}>
        <View style={{ flex: 1.5 }} />
        <Property title="" text={name} />
        <Property title="Abv: " text={abv} />
        <Property title="Ibu: " text={ibu} />
        <View style={{ flex: 1.5 }} />
      </View>
    </TouchableOpacity>
  );
};

export default RecommendedBeer;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center'
  },
  image: {
    flex: 0.5,
    aspectRatio: 1,
    margin: 10,
  },
  blankImage: {
    flex: 0.5,
    aspectRatio: 1,
    borderRadius: 5,
    margin: 10,
    backgroundColor: '#fafafa'
  },
  infoContainer: {
    flexDirection: 'column',
    justifyContent: 'center',
    flex: 1,
    marginLeft: deviceWidth * 0.05,
  },
})