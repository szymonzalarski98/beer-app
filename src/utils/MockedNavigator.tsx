import React, { ReactNode } from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
interface Props {
  component: ReactNode,
  params?: {}
}

const Stack = createStackNavigator();
const MockedNavigator: React.SFC<Props> = ({component, params = {}}) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="MockedScreen"
          component={component}
          initialParams={params}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default MockedNavigator;