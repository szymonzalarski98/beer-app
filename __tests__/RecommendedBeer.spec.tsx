import React from 'react';
import { render } from 'react-native-testing-library';
import MockedNavigator from '../src/utils/MockedNavigator';
import RecommendedBeer from '../src/components/details-card/RecommendedBeer';

describe('Test RecommendedBeer component', () => {
  const RecommendedBeerComponent = () => <RecommendedBeer id={1} name="Test" abv={10} ibu={11} image_url="test_image" />;
  it ('Should match snapshot', () => {
    const { baseElement, getByText } = render(<MockedNavigator component={RecommendedBeerComponent} />);
    expect(getByText('Test')).not.toBeNull();
    expect(getByText('10')).not.toBeNull();
    expect(getByText('11')).not.toBeNull();
    expect(baseElement).toMatchSnapshot();
  });
});