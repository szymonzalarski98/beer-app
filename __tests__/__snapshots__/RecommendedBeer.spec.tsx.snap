// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RecommendedBeer component Should match snapshot 1`] = `undefined`;
